### ISSUE TITLE
[java] MissingStaticMethodInNonInstantiatableClass false-positive for Lombok's @UtilityClass for classes with non-private fields

### ISSUE THREAD

**Issue by dawiddc** [pmd/pmd#4393]
**Affects PMD Version: 6.52.0**

**Rule: MissingStaticMethodInNonInstantiatableClass**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#missingstaticmethodinnoninstantiatableclass>

**Description:**

Issue #4224 fixed false-negative problem of omitting Lombok's @UtilityClass from being verified at all, but introduced problems for utility classes that contain only field members. Actually, we should check whether a class has both private fields only AND private methods only.

**Code Sample demonstrating the issue:**

```java
@UtilityClass
public final class Test {
   public static final String CONSTANT = "Constant";
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
This is actually one of the many things being fixed in #4370 as I revamped this rule for PMD7.

**Comment by dawiddc** [comment]
WDYT about merging this one to 6.x line and then overriding it with #4370 solution within 7.x line?
The pros are that we won't have to wait so long to have it fixed + there are few test cases added that might be useful within this PR.
